src/trans.rs

```diff
@@ -1310,25 +1337,12 @@ impl<'v, 'tcx: 'v, 'module: 'v> BinaryenFnCtxt<'v, 'tcx, 'module> {
     }
 
     fn trans_lval(&mut self, lvalue: &Lvalue<'tcx>) -> Option<BinaryenLvalue> {
+        let mir = self.mir.borrow();
+
         let i = match *lvalue {
-            Lvalue::Arg(i) => i.index() as u32,
-            Lvalue::Var(i) => {
+            Lvalue::Local(i) => {
```

Comment:
I think there's a little more that needs done here. I had a change that updated to nightly-2016-09-30 which touched this area. Basically, it seems like the different kinds of lvalues got folded into a local, which had its own kind field. See https://github.com/eholk/mir2wasm/commit/832a89316dbf0eba75d515c3611ad30091386574#diff-ed1805005a6effb6c15eeed860cdd597L1314 and https://github.com/eholk/mir2wasm/commit/832a89316dbf0eba75d515c3611ad30091386574#diff-ed1805005a6effb6c15eeed860cdd597L252

Of course, maybe it's that my change just made things more complicated than the need to be.